tests/unit/libbpf_test.cpp

```diff
@@ -3388,26 +3393,32 @@ _test_batch_iteration_maps(fd_t& map_fd, uint32_t batch_size, bpf_map_batch_opts
         REQUIRE(result == 0);
         // Number of entries retrieved (batch_size_count) should be less than or equal to requested_batch_size.
         REQUIRE(batch_size_count <= static_cast<uint32_t>(batch_keys.size()));
-        REQUIRE(batch_size_count <= static_cast<uint32_t>(batch_values.size()));
+        REQUIRE(batch_size_count <= static_cast<uint32_t>(batch_values.size()) / value_size);
 
         batch_keys.resize(batch_size_count);
-        batch_values.resize(batch_size_count);
+        batch_values.resize(batch_size_count * value_size);
         returned_keys.insert(returned_keys.end(), batch_keys.begin(), batch_keys.end());
         returned_values.insert(returned_values.end(), batch_values.begin(), batch_values.end());
 
         in_batch = &out_batch;
     }
 
     REQUIRE(returned_keys.size() == batch_size);
-    REQUIRE(returned_values.size() == batch_size);
-
-    std::sort(returned_keys.begin(), returned_keys.end());
-    std::sort(returned_values.begin(), returned_values.end());
+    REQUIRE(returned_values.size() == batch_size * value_size);
```

Comment:
nit: should it be `batch_size_count` instead of `batch_size`?

Reply:
here and also in the following code.

Reply:
>> should it be batch_size_count instead of batch_size?

No, it should be batch_size only, which is the total requested.

FYI, batch_size_count is used to retrieve requested_batch_size, where small blocks of key/value are retrieved in iteration.